src/shared/state/state.h

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef ADDONSTATE_H
+#define ADDONSTATE_H
+
+#include <QJsonObject>
+#include <QObject>
+
+#include "localstate.h"
+#include "sessionstate.h"
+
+/**
+ * @brief Struct containing the different types of state.
+ *
+ * These are:
+ *
+ * - `session` state: A type of state which is **not** persisted throughout
+ * device sessions.
+ * - `local` state: A type of state with is persisted throughout device
+ * sessions.
+ */
+struct State : public QObject {
+  Q_OBJECT
+  Q_PROPERTY(SessionState* session READ session)
+  Q_PROPERTY(LocalState* local READ local)
+
+ public:
+  State(QObject* parent, const SessionState& session, const LocalState& local)
+      : QObject(parent), m_session(session), m_local(local) {}
+
+  ~State() = default;
+
+  SessionState* session() { return &m_session; }
```

Comment:
const ?

Reply:
Hum, we need to be able to call non-const methods with this. Would adding `const` here prevent us from doing that?